bundles/org.openhab.binding.openthermgateway/src/main/java/org/openhab/binding/openthermgateway/internal/Message.java

```diff
@@ -149,7 +139,7 @@ public String toString() {
     }
 
     public Message(String code, MessageType messageType, int id, String data) {
```

Comment:
It is possible for `CodeType.valueOf(code);` to throw an exception if the given code isn't one of the enum values. Since a constructor isn't really a suitable place to do exception handling I think is would be better if you changed the `code` parameter from String to CodeType so that you push the burden of exception handling to the caller instead.

```suggestion
    public Message(CodeType code, MessageType messageType, int id, String data) {
```

Reply:
I've moved the evaluation of the CodeType into the parse() method, which uses a regexp to validate the code type is valid before parsing so won't error.